crates/iota-json-rpc/src/governance_api.rs

```diff
@@ -758,6 +761,36 @@ pub struct ValidatorExchangeRates {
     pub rates: Vec<(EpochId, PoolTokenExchangeRate)>,
 }
 
+/// Backfill missing rates for some epochs due to safe mode. If a rate is
+/// missing for epoch e, we will use the rate for epoch e-1 to fill it. Rates
+/// returned are in descending order by epoch.
+fn backfill_rates(
+    rates: Vec<(EpochId, PoolTokenExchangeRate)>,
+) -> Vec<(EpochId, PoolTokenExchangeRate)> {
+    if rates.is_empty() {
+        return rates;
+    }
+    let min_epoch = *rates.iter().map(|(e, _)| e).min().unwrap();
+    let max_epoch = *rates.iter().map(|(e, _)| e).max().unwrap();
+    let mut filled_rates = Vec::new();
+    let mut prev_rate = None;
+    for epoch in min_epoch..=max_epoch {
+        match rates.iter().find(|(e, _)| *e == epoch) {
+            Some((e, rate)) => {
+                prev_rate = Some(rate.clone());
+                filled_rates.push((*e, rate.clone()));
+            }
+            None => {
+                if let Some(rate) = prev_rate.clone() {
+                    filled_rates.push((epoch, rate));
+                }
+            }
+        }
+    }
```

Comment:
Same here:
```suggestion
    rates.sort_unstable_by_key(|(epoch_id, _)| *epoch_id);
    let mut filled_rates: Vec<(EpochId, PoolTokenExchangeRate)> = Vec::with_capacity(rates.len());
    let mut missing_rates = Vec::new();
    for (epoch_id, rate) in rates {
        if let Some((prev_epoch_id, prev_rate)) = filled_rates.last() {
            for missing_epoch_id in prev_epoch_id + 1..epoch_id {
                missing_rates.push((missing_epoch_id, prev_rate.clone()));
            }
        };
        filled_rates.append(&mut missing_rates);
        filled_rates.push((epoch_id, rate));
    }
```

Reply:
Thanks for the suggestions! I also make the rates to be `mut` in [4abb6d4](https://github.com/iotaledger/iota/pull/5184/commits/4abb6d4e03e0ae943664580f63c4649820655068).